src/lapack/backends/cusolver/cusolver_task.hpp

```diff
@@ -50,10 +50,13 @@ namespace cusolver {
 
 template <typename H, typename F>
 static inline void host_task_internal(H &cgh, sycl::queue queue, F f) {
+#ifdef SYCL_EXT_ONEAPI_ENQUEUE_NATIVE_COMMAND
+    cgh.ext_codeplay_enqueue_native_command([f, queue](sycl::interop_handle ih){
+#else
     cgh.host_task([f, queue](sycl::interop_handle ih) {
+#endif
         auto sc = CusolverScopedContextHandler(queue, ih);
         f(sc);
-        sc.wait_stream(queue);
```

Comment:
This is the only change that also affects the code logic when "SYCL_EXT_ONEAPI_ENQUEUE_NATIVE_COMMAND" is not defined. I removed it because it is a duplication of the calls to the SYNC macro that happens within lapack function implementations.

Reply:
Where does this `SYNC` macro get called? Does it know not to do so when using `enqueue_native_command`?

Reply:
mainly here: https://github.com/oneapi-src/oneMKL/pull/572/files#diff-b374cd674b06d7901c4ad5607b38b48de30a5f9e7215d58130ff80c7db51e680

and here: https://github.com/oneapi-src/oneMKL/pull/572/files#diff-bca3dc194e431af2adf1dcb0fdcd5bf8ba243163fdcac5bc304bf340e933d9d8